api/client/push.go

```diff
@@ -30,14 +28,6 @@ func (cli *DockerCli) CmdPush(args ...string) error {
 		return err
 	}
 
-	var tag string
-	switch x := ref.(type) {
-	case reference.Canonical:
-		return errors.New("cannot push a digest reference")
```

Comment:
oh you are right, something is missing in the `ImagePush` in remote API (same as `ImageTag` in fact)